src/bidiMapper/domains/script/Realm.ts

```diff
@@ -833,16 +812,176 @@ export class Realm {
     this.#realmStorage.knownHandlesToRealmMap.delete(handle);
   }
 
-  dispose() {
-    this.#eventManager.registerEvent(
-      {
-        type: 'event',
-        method: ChromiumBidi.Script.EventNames.RealmDestroyed,
-        params: {
-          realm: this.realmId,
+  dispose(): void {
+    for (const browsingContext of this.associatedBrowsingContexts) {
+      this.#eventManager.registerEvent(
+        {
+          type: 'event',
+          method: ChromiumBidi.Script.EventNames.RealmDestroyed,
+          params: {
+            realm: this.realmId,
+          },
         },
-      },
-      this.browsingContextId
+        browsingContext.id
+      );
+    }
+  }
+}
+
+export class WindowRealm extends Realm {
```

Comment:
Move to a separate file please

Reply:
We've ran into issues before in Puppeteer due to separating classes from their base because of circular imports. It's possible to separate the classes into a separate file, but you would need an entry point that takes care of the JS hoisting. Tl;dr, it's much harder for people to work with the code if we try separating them.

Reply:
If Realm does not depend on WindowRealm it should be no problem separating it.

Reply:
Even if it does not, I'd rather keep same classes together to prevent a potential problem in the future like the locators issue.